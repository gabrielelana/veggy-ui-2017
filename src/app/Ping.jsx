import React from 'react'
import request from 'superagent'

export default class Ping extends React.Component{
  constructor() {
    super()
    this.handlePingClick = this.handlePingClick.bind(this)
    this.state = {counter: 0}
  }
  componentDidMount() {
    const ws = new WebSocket('ws://localhost:4000/ws')
    ws.onopen = () => { 
      setInterval(() => ws.send('ping'), 5000)
    }
    ws.onmessage = (evt) => {
      const data = JSON.parse(evt.data)
      if (data.event === 'pong') {
        this.setState({ counter: data.counter })
      }
    }
  }
  handlePingClick() {
    request
      .post('http://localhost:4000/commands')
      .set('Content-Type', 'application/json')
      .send({command: 'Ping'})
      .end()
  }
  render() {
    return (
      <div>
        <h3>Ping!</h3>
        <button className="button" onClick={this.handlePingClick}>Ping</button>
        <p>Pongs received: {this.state.counter}</p>
      </div>
    )
  }
}