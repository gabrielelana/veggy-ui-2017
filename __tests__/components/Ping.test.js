import React from 'react'
import renderer from 'react-test-renderer'
import Ping from '../../src/app/Ping'

describe('<Ping />', () => {
  test('should render', () => {
    window.WebSocket = function(){}
    const component = renderer.create(<Ping />)
    expect(component.toJSON()).toMatchSnapshot()
  })
})