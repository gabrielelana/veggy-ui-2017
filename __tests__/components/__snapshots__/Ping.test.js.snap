// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Ping /> should render 1`] = `
<div>
  <h3>
    Ping!
  </h3>
  <button
    className="button"
    onClick={[Function]}
  >
    Ping
  </button>
  <p>
    Pongs received: 
    0
  </p>
</div>
`;
